fix(open-api-gateway): fix service name docs and remove unused modelDir option

cogwirrel opened the pull request:
Not counting this as a breaking change as it was already broken in https://github.com/aws/aws-prototyping-sdk/pull/280 (which was marked as breaking 😄 ).

cogwirrel committed:
fix(open-api-gateway): fix service name docs and remove unused modelDir option

agdimech reviewed (commented):
There are additional places in the README where serviceName is not present. Please add to all other locations.

agdimech commented:
On a seperate note: Can we disable jest for the GeneratedTypescriptClientProject given no tests are present?

cogwirrel committed:
feat(open-api-gateway): disable jest for generated typescript client unless explicitly enabled

Since there are no tests in the generated typescript client, we disable jest by default.
